File: src/main/java/org/example/keycloak/schemas/ScimListResponse.java
package org.example.keycloak.schemas;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public record ScimListResponse<T>(

    List<String> schemas,
    int totalResults,
    int startIndex,
    int itemsPerPage,
    @JsonProperty("Resources")
    List<T> resources

) {

}

File: src/main/kotlin/org/example/keycloak/schemas/ScimListResponse.kt
package org.example.keycloak.schemas

import com.fasterxml.jackson.annotation.JsonProperty

data class ScimListResponse<T>(
    val schemas: List<String>,
    val totalResults: Int,
    val startIndex: Int,
    val itemsPerPage: Int,
    @JsonProperty("Resources")
    val resources: List<T>
)